File: backend/src/models/assistencia.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class AssistenciaCreate(BaseModel):
    acidente_id: str
    tipo: str
    latitude_atual: Optional[float] = None
    longitude_atual: Optional[float] = None


class AssistenciaUpdate(BaseModel):
    status: Optional[str] = None
    latitude_atual: Optional[float] = None
    longitude_atual: Optional[float] = None


class AssistenciaResponse(BaseModel):
    assistencia_id: str
    acidente_id: str
    tipo: str
    status: str
    latitude_atual: Optional[float] = None
    longitude_atual: Optional[float] = None
    hora_inicio: datetime
    hora_fim: Optional[datetime] = None

File: backend/src/models/Assistencia.js
const mongoose = require('mongoose');

const assistenciaSchema = new mongoose.Schema({
  assistencia_id: { type: String },
  acidente_id: { type: String },
  tipo: { type: String, default: 'AMBULANCIA' },
  status: { type: String, default: 'A_CAMINHO' },
  latitude_atual: { type: Number },
  longitude_atual: { type: Number },
  hora_inicio: { type: Date, default: null },
  hora_fim: { type: Date, default: null }
}, { collection: 'assistencias', timestamps: false, strict: false });

module.exports = mongoose.model('Assistencia', assistenciaSchema);
